pennylane/devices/qubit/simulate.py

```diff
@@ -44,6 +45,30 @@ def measure_state_diagonalizing_gates(
     return measurementprocess.process_state(qml.math.flatten(state), wires)
 
 
+def measure_hamiltonian_expval(
+    measurementprocess: MeasurementProcess, state: TensorLike
+) -> TensorLike:
+    """Measure the expectation value of the state when the measured observable is a ``Hamiltonian`` or
+    ``SparseHamiltonian``.
+
+    Args:
+        measurementprocess (MeasurementProcess): measurement process to apply to the state
+        state (TensorLike): the state to measure
+
+    Returns:
+        TensorLike: the result of the measurement
+    """
+    # Add case for backprop later
+    Hmat = measurementprocess.obs.sparse_matrix()
+    _state = qml.math.toarray(state)
```

Comment:
Shouldn't `_state` already be an array? Any idea what this is for?

Reply:
I'm not entirely sure, but my intuition is that it was for casting the state to a numpy array for passthrough devices since the `csr_matrix` only accepts `ndarray` inputs